src/AzureAppConfigurationImpl.ts

```diff
@@ -45,6 +52,9 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
     #featureFlagRefreshInterval: number = DEFAULT_REFRESH_INTERVAL_IN_MS;
     #featureFlagRefreshTimer: RefreshTimer;
 
+    // selectors
+    #featureFlagSelectors: PagedSettingSelector[] = [];

```

Comment:
Note: Here for each selector, both etag and corresponding items of each page are cached. The benefit is, when one of the pages is changed, we can get items of unchanged pages (without sending extra requests when 304) and do the composition among selectors.

But, this leads to more memory consumption. Since configuration/feature flags don't change very often,  in most requests it would be 304. Discussed with Zhenlan, and we can only cache the page etags without the items. When attempting to refresh, if all pages are 304, we do nothing. If any of the page is 200, we send requests again forcing to listing all the pages, and update the cached page etags. I will update the PR to go with this approach.

Reply:
SDK v1.6.0 has a bug, when nothing changes, on listing configuration settings with pageEtags, only the first page.status_code is 304, all the other pages are 200.
Created an issue to track: https://github.com/Azure/azure-sdk-for-js/issues/30282

But the bug doesn't affects our impl in this PR, and actually not much impact, as in most of our use cases there should be only one page i believe. Just FYI.

Reply:
SDK v1.6.1 is out carrying the fix.